app/src/main/java/org/mozilla/focus/tiles/HomeTilesManager.kt

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.tiles
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import android.content.SharedPreferences
+import org.json.JSONArray
+import org.json.JSONObject
+
+const val HOME_TILES_PREFS = "hometilesPrefs"
+const val CUSTOM_SITES_LIST = "customSitesList"
+
+const val KEY_URL = "url"
+const val KEY_TITLE = "title"
+const val KEY_IMG = "img"
+const val KEY_ID = "identifier"
+
+object HomeTilesManager {
+    fun pinSite(context: Context, url: String) {
```

Comment:
nit: Can this be called by multiple threads? If not, annotate it. If so, make sure the get-and-set is atomic (either with locking or a data structure that makes it possible). Also if so, please reflag me for review.

Reply:
Should only be called from a user action (e.g., toggling the pin button, or maybe in the future deleting from the home screen, which will be UI thread)